File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int size = nums.length;
        int [] result = new int[size];
        int index =0;
          int product =1;
        for(int i=0;i<size;)
        {
            int j =0;
            while(j<size){
                    if(i==j){
                    // do nothing
                    }
                    else
                    {
                        product *= nums[j];
                    }
                    j++;
            }
             
            
            result[i]=product;
            i++;
            index =i;
            product =1;
        }
        
        return result;
    
}  
}

File: Data Structures & Algorithms/subsets/submission-0.kt
class Solution {
    fun subsets(nums: IntArray): List<List<Int>> {
        // main list that get all sub sets

        val result = mutableListOf<List<Int>>()

        // all subset solution sould contains empty subset
        result.add(emptyList())


   for(num in nums){
        val subsets = mutableListOf<List<Int>>()

     for(value in result){
            val subset = value.toMutableList()
            subset.add(num)
            subsets.add(subset)
     }

     result.addAll(subsets)

        }
        return result
    }
}
